syntax = "proto3";

package aggregator_proto;

// Every point coordinate (ie: base field element)
// is treated as an hexadecimal string.
// The points in G1 are represented in affine form.
message HexPointBaseGroup1Affine {
    // First coordinate of the point
    string x_coord = 1;
    // Second coordinate of the point
    string y_coord = 2;
}

// The points in G2 are represented in affine form.
//
// IMPORTANT:
// Here we assume that the twist field elements are polynomials of degree 1
// i.e. they can be represented using a vector of 2 base field elements.
message HexPointBaseGroup2Affine {
    // First coordinate of the point
    string x_c1_coord = 1;
    string x_c0_coord = 2;
    // Second coordinate of the point
    string y_c1_coord = 3;
    string y_c0_coord = 4;
}